randovania/generator/hint_distributor.py

```diff
@@ -39,12 +36,91 @@
 
 HintProvider = Callable[[PlayerState, GamePatches, Random, PickupIndex], LocationHint | None]
 
+HintTargetPrecision = tuple[PickupIndex, PrecisionPair]
 
-def _not_empty(it: Iterator) -> bool:
-    return sum(1 for _ in it) > 0
+HintFeatureGaussianParams = tuple[float, float]
+"""The mean and standard deviation defining a Gaussian distribution."""
 
 
-HintTargetPrecision = tuple[PickupIndex, PrecisionPair]
+class FeatureChooser[FeatureT: HintFeature, PrecisionT: Enum]:
```

Comment:
Hm... not sure about bumping Rdv's minimum to 3.12
Maaaybe it's fine? Hmm

Though if that does happen, make sure to bump `requires-python` in `pyproject.toml`

Reply:
yeah, I didn't think so either. so the 3.12 syntax was the only solution I found to binding a typevar to DataclassInstance